data_structures/src/data_request.rs

```diff
@@ -357,6 +356,27 @@ pub fn calculate_tally_change(
         + dr_output.commit_fee * (witnesses - commits_count)
 }
 
+pub fn calculate_reward(
+    reveals_count: usize,
+    slashed_count: usize,
+    honests_count: usize,
+    dr_output: &DataRequestOutput,
+) -> (u64, u64) {
+    if reveals_count > 0 {
+        let slashed_count = slashed_count as u64;
+        let honests_count = honests_count as u64;
+        let slashed_collateral_reward = dr_output.collateral * slashed_count / honests_count;
```

Comment:
`honests_count` can be 0?

Reply:
In case of reveals_count > 0 it is not possible

Reply:
What if everyone lies?

Reply:
Our honests people is people in consensus. If all lies are in consensus, so all of them are honest people for us